### ISSUE TITLE
Site editor: Tracks which record block insertion are being fired by navigating the site editor

### ISSUE THREAD

**Issue by p-jackson** [Automattic/wp-calypso#79818]
### Quick summary

The `wpcom_block_inserted` and `wpcom_block_picker_block_inserted` events are used to track block insertions/replacements in the editor. These events have a `insert_method` prop which records what method was used to insert the block (e.g. `quick-inserter`).

These events are currently being fired by the site editor even when blocks aren't being inserted. It makes it impossible to use the `insert_method` to reason about which insertion methods are being used by the user. https://github.com/Automattic/wp-calypso/issues/50426#issuecomment-1648841001

I think this should be done as part of dotcom engineering maintenance work.

### Steps to reproduce

1. Open the site editor
2. Open the network tab in the dev tools and filter by `wpcom_block_inserted`
3. Navigate around the sidebar. I reproduced the issue by clicking "patterns" and then clicking the back button to go back to "design"

### What you expected to happen

`wpcom_block_inserted` should not be recorded while navigating.

### What actually happened

`wpcom_block_inserted` events are recorded while navigating.

### Impact

Most (> 50%)

### Available workarounds?

No but the platform is still usable

**Comment by cuemarie** [comment]
📌 ACTIONS
- Due to the age of this issue, I've downgraded the priority from `High` to `Normal` to better reflect its status in the larger context of open bugs in this repo.

**Comment by gabrielcaires** [comment]
New Triage
I could reproduce the bug, 

I opened the site editor and navigated the patterns and other interface elements, and I could only see the event when it happened. 

I will set as need triage to have a double checking.

**Comment by Robertght** [comment]
This is what I can see on my end:

![Image](https://github.com/user-attachments/assets/7593c458-15f2-4676-9ba0-432082ac5c60)

@gabrielcaires Is this enough to confirm the presence of the issue?

**Comment by mmtr** [comment]
I can still replicate the issue. Since the "Site Editor Experience" is an area assigned to @Automattic/t-rex and @Automattic/lego in PdqkMK-1nV-p2, maybe some of us can take a look a it.

**Comment by ntsekouras** [comment]
Has this been resolved by: https://github.com/Automattic/wp-calypso/pull/100543? -cc @okmttdhr

**Comment by okmttdhr** [comment]
Yup, I can't replicate the issue anymore. I'm closing the issue.
